src/main/java/org/jenkinsci/plugins/ssegateway/sse/EventDispatcher.java

```diff
@@ -67,7 +68,11 @@
 public abstract class EventDispatcher implements Serializable {
 
     public static final String SESSION_SYNC_OBJ = "org.jenkinsci.plugins.ssegateway.sse.session.sync";
-    private static final Logger LOGGER = Logger.getLogger(EventDispatcher.class.getName());
+    private static final Logger LOGGER = LoggerFactory.getLogger( EventDispatcher.class.getName());
+
+    private static final ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(
+        Integer.getInteger( EventDispatcher.class.getName() + ".scheduledExecutorService.size", 4 ),
+        r -> new Thread( "EventDispatcher.retryProcessor" ));
```

Comment:
Wait a second, this thread factory completely ignores the runnable `r`, so it just starts a thread with no actual task to run, which means that we never retry anything, right? I think you meant something like this:

```suggestion
        r -> new Thread( r, "EventDispatcher.retryProcessor" ));
```
But this brings me back to what I mentioned in https://github.com/jenkinsci/sse-gateway-plugin/pull/35/files#r311076879, and if users tried this out and things worked correctly, then maybe all of the retry handling should just be deleted and instead we would just never retry anything?

Reply:
Or maybe this is only a problem for the first 4 tasks, and after that the tasks use the existing threads and run the submitted tasks directly?

Reply:
yup I'm not sure if we really need this retry.
To not break everything I was thinking to make it optional via a sys props so if this doesn't work users can reactivate it?

Reply:
Yeah, or maybe make it so a particular event can only be retried once or something? I wonder whether the events being retried even matter at all. @kshultzCB pointed out to me that the Blue Ocean ATH makes a lot of use of the SSE events, so if you did disable to retries by default, it would be good to open a Blue Ocean PR that uses a snapshot of this PR to make sure it doesn't regress any tests.

Reply:
I'm trying but not BO ci is not working very well https://ci.blueocean.io/job/blueocean/job/JENKINS-58684-sse-gateway-1.20-SNAPSHOT/ :(

Reply:
dohhhh blue build here https://ci.blueocean.io/job/blueocean/job/JENKINS-58684-sse-gateway-1.20-SNAPSHOT/4/ :)

Reply:
@dwnusbaum at this stage I'm tempted to merge as is. release and wait for report from users as logging of issues has been improved. WDYT?